src/LinuxTracing/UprobesUnwindingVisitor.cpp

```diff
@@ -214,52 +217,93 @@ UprobesUnwindingVisitor::ComputeCallstackTypeFromStackSample(
   return Callstack::kComplete;
 }
 
-void UprobesUnwindingVisitor::Visit(uint64_t event_timestamp,
-                                    const StackSamplePerfEventData& event_data) {
+template <typename StackPerfEventDataT>
+bool UprobesUnwindingVisitor::UnwindStack(const StackPerfEventDataT& event_data,
+                                          Callstack* callstack) {
   ORBIT_CHECK(listener_ != nullptr);
   ORBIT_CHECK(current_maps_ != nullptr);
 
-  return_address_manager_->PatchSample(event_data.tid, event_data.GetRegisters().sp,
+  return_address_manager_->PatchSample(event_data.GetCallstackTid(), event_data.GetRegisters().sp,
                                        event_data.GetMutableStackData(), event_data.GetStackSize());
 
   StackSliceView event_stack_slice{event_data.GetRegisters().sp, event_data.GetStackSize(),
-                                   event_data.data.get()};
+                                   event_data.GetStackData()};
   std::vector<StackSliceView> stack_slices{event_stack_slice};
-
-  const auto& stream_id_to_user_stack = thread_id_stream_id_to_stack_slices_.find(event_data.tid);
+  const auto& stream_id_to_user_stack =
+      thread_id_stream_id_to_stack_slices_.find(event_data.GetCallstackTid());
   if (stream_id_to_user_stack != thread_id_stream_id_to_stack_slices_.end()) {
     for (const auto& [unused_stream_id, user_stack_slice] : stream_id_to_user_stack->second) {
       stack_slices.emplace_back(user_stack_slice.start_address, user_stack_slice.size,
                                 user_stack_slice.data.get());
     }
   }
 
-  LibunwindstackResult libunwindstack_result = unwinder_->Unwind(
-      event_data.pid, current_maps_->Get(), event_data.GetRegistersAsArray(), stack_slices);
+  LibunwindstackResult libunwindstack_result =
+      unwinder_->Unwind(event_data.GetCallstackPidOrMinusOne(), current_maps_->Get(),
```

Comment:
I think we should add a comment explaining that -1 is fine passed to the unwinder. Basically what you told me, that it's only needed for reading the process memory, etc.

I would also specify that theoretically we could have that information from `SwitchesStatesNamesVisitor` (see `SwitchesStatesNamesVisitor::GetPidOfTid`). A TODO seems appropriate.
This by the way reinforces my point that we need a central place for processing, as we are just ignoring the possibility as it would be a bit too messy to fix it.

Reply:
Done.